docs/src/pages/docs/events.md

```diff
@@ -0,0 +1,200 @@
+        emit!(MyEvent {
+            data: 5,
+            label: "hello".to_string(),
+        });
+        Ok(())
+    }
+
+// ...
+#[event]
+pub struct MyEvent {
+    pub data: u64,
+    #[index]
+    pub label: String,
+}
+
+```
+
+Observe that we are using the event to provide a return value to a successful call to `initialize`.
+The data returned is fairly trivial, a `u64` integer and a short `String`. Without the event, we would have no specific information about the outcome of the initialize function, besides that it was successful.
+
+Note that the event is declared with the `#[event]` macro, and defined using a `struct`. To supply the values,
+the pattern `emit!(MyEvent {...});` is used. The code above is a basic stamp that can be used for creating and consuming events within your program.
+
+Next, nothing fancy but we'll add a new function called `test_event` with a new event that emits some other values. Complete code for `src/lib.rs` is shown below:
+
+```rust
+use anchor_lang::prelude::*;
+
+declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
+
+#[program]
+pub mod events {
+    use super::*;
+
+    pub fn initialize(_ctx: Context<Initialize>) -> Result<()> {
+        emit!(MyEvent {
+            data: 5,
+            label: "hello".to_string(),
+        });
+        Ok(())
+    }
+
+    pub fn test_event(_ctx: Context<TestEvent>) -> Result<()> {
+        emit!(MyOtherEvent {
+            data: 6,
+            label: "bye".to_string(),
+        });
+        Ok(())
+    }
+}
+
+#[derive(Accounts)]
+pub struct Initialize {}
+
+#[derive(Accounts)]
+pub struct TestEvent {}
+
+#[event]
+pub struct MyEvent {
+    pub data: u64,
+    #[index]
+    pub label: String,
+}
+
+#[event]
+pub struct MyOtherEvent {
+    pub data: u64,
+    #[index]
+    pub label: String,
+}
+```
+
+## Consuming Events
+
+To test our program, we'll build a test client that will be able to consume the events. The structure for doing this is essentially:
+```javascript
+    // ...
+    let listener = null;
+
+    let [event, slot] = await new Promise((resolve, _reject) => {
+      listener = program.addEventListener("MyEvent", (event, slot) => {
+        resolve([event, slot]);
+      });
+      program.rpc.initialize();
+    });
+    await program.removeEventListener(listener);
+    // ...
+```
+
+We set up a `listener` that will catch when the event is emitted by the program. Since it's an asynchronous activity, we must include `await` to receive our return values. We call the first function using `program.rpc.initialize()` and when the event happens, we destructure the output into variables `event` and `slot`. Finally, when it's done, we close the listener. We can access our output values as follows: `event.data.toNumber()` or `event.label`.
+
+With the explanation out of the way, edit the file in `tests/events.ts`. Copy-paste the test code into the file as follows, then run `$ anchor test` from the command line to verify that it's working as intended:
```

Comment:
Could you add a line about *what* the expected output to this is?

Reply:
Done.